src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -874,25 +874,23 @@ private BoundExpression MakeNullableHasValue(SyntaxNode syntax, BoundExpression
             //
             // tempx = x; 
             // tempy = y;
-            // result = tempx.GetValueOrDefault() == tempy.GetValueOrDefault() ? 
-            //          tempx.HasValue == tempy.HasValue : 
-            //          false;
+            // result = (tempx.GetValueOrDefault() == tempy.GetValueOrDefault()) &
+            //          (tempx.HasValue == tempy.HasValue);
```

Comment:
> Only comparison operators end up here and the result is `bool`.

I meant XOR, not the C# operator. As in cases where `tempx` is nullable or `tempy` is nullable, but not both.

Reply:
Looks like there is not (or I couldn't find it). Will add a test to verify new codegen.

Reply:
I do not think this change has any effect on `^`, since it is not a comparison operator.
It is worth investigating whether the same optimization is applicable to `^`, but I assume that will come as another change.

Reply:
Ah, I see. you meant the operands differing in being non-nullable. 
Yes, it is worth adding a test.